src/realm/error_codes.cpp

```diff
@@ -60,11 +68,56 @@ StringData ErrorCodes::error_string(Error code)
         case ErrorCodes::WebSocket_TLSHandshakeFailed:
             return "WebSocket: TLS Handshake Failed";
 
+        /// WebSocket Errors - reported by server
+        case ErrorCodes::WebSocket_Unauthorized:
+            return "WebSocket: Unauthorized";
+        case ErrorCodes::WebSocket_Forbidden:
+            return "WebSocket: Forbidden";
+        case ErrorCodes::WebSocket_MovedPermanently:
+            return "WebSocket: Moved Permanently";
+        case ErrorCodes::WebSocket_Client_Too_Old:
+            return "WebSocket: Client Too Old";
+        case ErrorCodes::WebSocket_Client_Too_New:
+            return "WebSocket: Client Too New";
+        case ErrorCodes::WebSocket_Protocol_Mismatch:
+            return "WebSocket: Protocol Mismatch";
+
         case ErrorCodes::UnknownError:
             [[fallthrough]];
         default:
             return "UnknownError";
     }
 }
 
+namespace {
```

Comment:
Can we move the close status error category and make_error_code out into a noinst part of the sync codebase? We really shouldn't conflate realm::ErrorCodes with std::error_code - I know they have basically the same name, but I really don't want us to start mixing them.

Reply:
We can. I think this will all go away once exception unification is in anyway

Reply:
I think it actually should be in sync::network. I moved it from webscoket.hpp/.cpp, so maybe it wasn't the best idea. I'll move it back since it makes more sense.